ChessBoard.Api/Controllers/ChessBoardController.cs

```diff
@@ -0,0 +1,68 @@
+using ChessBoard.Api.Entities;
+using ChessBoard.Api.Repositories;
+using ChessBoard.Api.Services;
+using Microsoft.AspNetCore.Mvc;
+using Serilog;
+
+
+namespace ChessBoard.Api.Controllers
+{
+    [ApiController]
+    [Route("chessboards")]
+    public class ChessBoardController : ControllerBase
+    {
+        private readonly IChessBoardRepository _chessBoardRepository;
+
+        public ChessBoardController(IChessBoardRepository chessBoardRepository)
+        {
+            _chessBoardRepository = chessBoardRepository;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<ChessBoardEntity>> GetAsync()
+
+        {
+            var allChessboards = await _chessBoardRepository.GetAllAsync();
+            return allChessboards.OrderBy(chessBoard => chessBoard.GeneratedChessBoard.Length);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<ChessBoardEntity>> GetChessboard(string id)
+        {
+            var chessBoard = await _chessBoardRepository.GetAsync(id);
+            return chessBoard is not null ? chessBoard : NotFound();
+        }
+
+
+        [HttpPost("{input}")]
+        public async Task<ActionResult> CreateAsync([FromRoute] string input)
+        {
+            var logger = new LoggerConfiguration().CreateLogger();
+
+            var chessBoardGenerator = new ChessBoardGenerator(logger);
```

Comment:
Before adjusting please provide a short plan on what steps you would take :)

Reply:
Hey, so do before I even try to come up with a plan, I am struggling to understand a bit, do you mean the logic should be put elsewhere? Because 100% we do not edit the individual `InputValidation` and `ChessBoardGenerator` classes/methods right? (to always have them as reusable). So if the logic should not happen in in controller, where should it happen? Or am I misunderstanding and you are talking about introducint DTOs maybe? :D

Reply:
Well, as a quick thought it could be something like `ChessBoardGeneratorService`, that inside uses:
1) InputValidator, possibly as constructor injected entity. 
2) And chess board generation logic to generate the chessboard - could be using `ChessBoardGenerator` or move logic from it to service itself
3) Repository, which would be responsible for saving the chessboard to database

Then controller would only call `ChessBoardGeneratorService.Create` and service would return result to the controller and then it would be pushed back to the ui. Something like that. If you have any similar questions let's discuss in slack ;)

Reply:
Understood, I think I should try adding the logic you described and let's review in next PR?

Reply:
Sounds good.